shopinvader_sale_coupon/services/cart.py

```diff
@@ -34,6 +42,26 @@ def _apply_coupon(self, cart, params):
             .process_coupon()
         )
 
+    def _remove_coupon(self, cart, params):
+        code = params["code"]
+        # First we check if this code is a coupon
+        coupon = self.env["coupon.coupon"].search([("code", "=", code)])
+        if coupon:
+            program = coupon.program_id
+        else:
+            # If it's not a coupon, it could be a promo with a code
+            program = self.env["coupon.program"].search([("promo_code", "=", code)])
```

Comment:
You can probably do this in a single search
```suggestion
        program = self.env["coupon.program"].search(
            [
                "|",
                ("promo_code", "=", code),
                ("coupon_ids.code", "=", code),
            ],
            limit=1,
        )
```